cmd/config/internal/commands/cmdset.go

```diff
@@ -93,12 +102,17 @@ func (r *SetRunner) preRunE(c *cobra.Command, args []string) error {
 	if setterVersion == "v2" {
 		var err error
 		r.Set.Name = args[1]
-		r.Set.Value = args[2]
+		if !valueSetFromFlag {
+			r.Set.Value = args[2]
+		}
 
 		// set remaining values as list values
-		if len(args) > 3 {
+		if len(args) > 2 && valueSetFromFlag {
```

Comment:
This should probably be an error case, users should either set as args or flags, not both at the same time.  It's unclear what the ordering would be if both were specified.

Reply:
Done, and test added.